src/CompositeRheologies.jl

```diff
@@ -206,13 +225,136 @@ function compute_εII_AD(v::Parallel, τII, args; tol=1e-6, verbose=false)
     return local_iterations_τII_AD(v, τII, args; tol=tol, verbose=verbose)
 end
 
-# COMPUTE DEVIATORIC STRESS
+#COMPUTE VOLUMETRIC STRAIN-RATE
+
+"""
+    compute_εvol(v::CompositeRheology{T,N}, p, args; tol=1e-6, verbose=false, n=1)
+
+Computing `εvol` as a function of `p` for a composite element is the sum of the individual contributions
+"""
+@generated  function compute_εvol(
+    v::CompositeRheology{T,N}, 
+    p::_T, 
+    args; 
+    tol=1e-6, verbose=false
+) where {T,_T,N}
+    quote
+        Base.@_inline_meta
+        εvol = zero(_T)
+        Base.Cartesian.@nexprs $N i ->
+            if isvolumetric(v.elements[i])
+                εvol += compute_εvol(v.elements[i], p, args)
+            end    
+        return εvol
+    end
+end
+
+
+@generated  function compute_εvol(
+    v::CompositeRheology{T,N}, 
+    p::Quantity, 
+    args; 
+    tol=1e-6, verbose=false
+) where {T,_T,N}
+    quote
+        Base.@_inline_meta
+        εvol = 0/s
+        Base.Cartesian.@nexprs $N i ->
+            if isvolumetric(v.elements[i])
+                εvol += compute_εvol(v.elements[i], p, args)
+            end
+        return εvol
+    end
+end
+
+# COMPUTE DEVIATORIC STRESS AND PRESSURE
 function compute_τII(v::CompositeRheology{T,N,0}, εII, args; tol=1e-6, verbose=false) where {T,N}
     # A composite rheology case with no parallel element; iterations for τII
     τII = local_iterations_εII(v, εII, args; tol=tol, verbose=verbose)
     return τII
 end
 
+"""
+    p,τII = compute_p_τII(v::CompositeRheology, εII, εvol, args; tol=1e-6, verbose=false) 
+
+This updates pressure `p` and deviatoric stress invariant `τII` in case the composite rheology structure has volumetric components, but does not contain plastic or parallel elements.
+The 'old' pressure should be stored in `args` as `args.P_old`   
+"""
+function compute_p_τII(
+    v::CompositeRheology{T,N,
+                    Npar,is_parallel,
+                    Nplastic,is_plastic,
+                    Nvol,is_vol,
+                    false}, 
+        εII::_T, 
+        εvol::_T,
+        args; 
+        tol=1e-6, verbose=false
+    ) where {T, N, _T, Npar, is_parallel, Nplastic, is_plastic, Nvol, is_vol}
+
+    # A composite rheology case that may have volumetric elements, but the are not 
+    # tightly coupled, so we do NOT perform coupled iterations.
+    τII = local_iterations_εII(v, εII, args; tol=tol, verbose=verbose)
+    P   = local_iterations_εvol(v, εvol, args; tol=tol, verbose=verbose)
+
+    return P,τII
+end
+
+
+"""
+    p,τII = compute_p_τII(v::CompositeRheology, εII, εvol, args; tol=1e-6, verbose=false) 
+
+This updates pressure `p` and deviatoric stress invariant `τII` in case the composite rheology structure has no volumetric elemnts, but may contain plastic or parallel elements. 
+In that case, pressure is not updated (`args.P` is used instead).     
+"""
+function compute_p_τII(
+    v::CompositeRheology{T,N,
```

Comment:
As a matter of format and style, align `v` with the arguments below and keep its type signature in a single line.

Reply:
agree with then aligning, but keeping the type signature in a single line makes it less readable in my opinion.